packages/api-explorer/src/components/SelectorContainer/ApiSpecSelector.tsx

```diff
@@ -49,7 +50,7 @@ export const ApiSpecSelector: FC<ApiSpecSelectorProps> = ({ spec }) => {
 
   const handleChange = (specKey: string) => {
     const matchPath = location.pathname.replace(`/${spec.key}`, `/${specKey}`)
-    history.push(matchPath)
+    navigate(matchPath, history)
```

Comment:
I would expect `history` first, but as Joseph mentioned, a custom hook should be implemented as a follow-up to clean this up.